common/src/test/java/com/revenuecat/purchases/backend_integration_tests/ProductionBackendIntegrationTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.revenuecat.purchases.backend_integration_tests
+
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.common.networking.Endpoint
+import com.revenuecat.purchases.common.offlineentitlements.ProductEntitlementMapping
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class ProductionBackendIntegrationTest: BaseBackendIntegrationTest() {
+    override fun apiKey() = Constants.apiKey
+
+    @Test
+    fun `can perform product entitlement mapping backend request`() {
+        var error: PurchasesError? = null
+        ensureBlockFinishes { latch ->
+            backend.getProductEntitlementMapping(
+                onSuccessHandler = { productEntitlementMapping ->
+                    assertThat(productEntitlementMapping.mappings.size).isEqualTo(34)
+                    assertThat(productEntitlementMapping.mappings["annual_freetrial"]).isEqualTo(
+                        ProductEntitlementMapping.Mapping(
+                            productIdentifier = "annual_freetrial",
+                            basePlanId = "p1y",
+                            entitlements = listOf("pro_cat")
+                        )
+                    )
+                    latch.countDown()
+                },
+                onErrorHandler = {
+                    error = it
+                    latch.countDown()
+                }
+            )
+        }
+        assertThat(error).isNull()
+        verify(exactly = 1) {
+            // Verify we save the backend response in the shared preferences
+            sharedPreferencesEditor.putString("/v1${Endpoint.GetProductEntitlementMapping.getPath()}", any())
+        }
```

Comment:
Note how these verifications are not in the load shedder tests. This is because the load shedder doesn't support etags, so since it doesn't return an etag, we won't be saving anything